### PR TITLE
feat(moment): impl moment api

### PR DESCRIPTION
**Pull request by gary02**
resolve https://github.com/thematters/matters-server/issues/3953

### CODE REVIEW

File: src/connectors/journalService.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    const content = sanitizeHTML(data.content)
+    const journal = await this.models.create({
+      table: 'journal',
+      data: {
+        authorId: user.id,
+        content: content,
+        state: JOURNAL_STATE.active,
+      },
+    })
+    if (data.assetIds && data.assetIds.length > 0) {
```
Comment (robertu7) [inline src/connectors/journalService.ts:45]: check asset author and type also?

File: src/types/journal.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { AUTH_MODE, NODE_TYPES } from 'common/enums'
+
+export default /* GraphQL */ `
+  extend type Mutation {
+    putJournal(input: PutJournalInput!): Journal! @auth(mode: "${AUTH_MODE.oauth}") @rateLimit(limit:2, period:300)
```
Comment (robertu7) [inline src/types/journal.ts:5]: confirm with product or CC on rate limiting?

File: src/queries/journal/comments.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  })
+
+  const where: CommentFilter = {
+    type: COMMENT_TYPE.journal,
+    targetId: id,
+    targetTypeId,
+  }
+  if (rest.filter) {
+    const { parentComment, author, state } = rest.filter
+    if (parentComment || parentComment === null) {
```
Comment (robertu7) [inline src/queries/journal/comments.ts:48]: can ignore `parentComment` for journal comments?

**Review by robertu7 — APPROVED** [review]

File: src/connectors/cloudflare/index.ts [hunk 4]
```diff
@@ -65,7 +64,11 @@ export class CloudflareService {
     return key
   }
 
-  baseUploadFile = async (folder: GQLAssetType, upload: any, uuid: string) => {
+  public baseUploadFile = async (
+    folder: GQLAssetType,
+    upload: any,
```
Comment (byhow) [inline src/connectors/cloudflare/index.ts:69]: not a big fan whenever it comes to `any`. We can explicitly type it with just the types we will be using here:
`const { createReadStream, mimetype, filename } = upload`

File: src/connectors/journalService.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            data: { assetId, journalId: journal.id },
+          })
+        )
+      )
+    }
+    return journal
+  }
+
+  public delete = async (id: string, user: User) => {
+    if (![USER_STATE.active, USER_STATE.banned].includes(user.state as any)) {
```
Comment (byhow) [inline src/connectors/journalService.ts:59]: ditto on using `any`

File: src/connectors/journalService.ts [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+      targetId: user.id,
+    })
+    if (isBlocked) {
+      throw new ForbiddenError(`user ${id} is blocked by target author`)
+    }
+    return this.models.upsert({
+      table: 'action_journal',
+      where: { targetId: id, userId: user.id },
+      create: { targetId: id, userId: user.id, action: 'like' },
+      update: {},
```
Comment (byhow) [inline src/connectors/journalService.ts:108]: can we omit this update : {}?

File: src/connectors/journalService.ts (hunk 1 above)
Comment (gary02) [reply src/connectors/journalService.ts:71]: so as content length

File: src/types/journal.ts (hunk 2 above)
Comment (gary02) [reply src/types/journal.ts:5]: just a temp value by now,  will  be update in future PR

**Pull request merged**